### PR TITLE
BSS round 2, hud_element.c funcs

### PR DESCRIPTION
**Pull request by ethteck**

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀🚀🚀🚀🚀🚀🚀 This PR matches 7 functions (+0.11%) on `us`.

✅ This PR fixes 4 warnings!

File: src/2e230_len_2190.c [hunk 1]
```diff
@@ -500,6 +502,7 @@ BGMPlayer* func_80054248(u8 arg0) {
             return D_8009A640;
         default:
             return NULL;
+            break;
```
Comment (nanaian) [inline src/2e230_len_2190.c:505]: I don't think we need a break after returns

File: src/world/area_end/end_00/DF6A20.c [hunk 2]
```diff
@@ -42,7 +52,29 @@ ApiStatus func_80242558_DF8F58(Evt* script, s32 isInitialCall) {
...
+    End18* end18 = &D_80244D74_DFB774[npcID];
+    NpcBlueprint bp;
+    Npc* npc;
+
+    bp.flags = 0x100;
+    bp.initialAnim = end18->initialAnim;
+    bp.onUpdate = NULL;
+    bp.onRender = NULL;
+
+    spr_allocateBtlComponentsOnWorldHeap = 1;
```
Comment (nanaian) [inline src/world/area_end/end_00/DF6A20.c:67]:
```suggestion
    spr_allocateBtlComponentsOnWorldHeap = TRUE;
```

File: src/world/area_end/end_00/DF6A20.c [hunk 3]
```diff
@@ -42,7 +52,29 @@ ApiStatus func_80242558_DF8F58(Evt* script, s32 isInitialCall) {
...
+    Bytecode* args = script->ptrReadPos;
+    s32 npcID = evt_get_variable(script, *args++);
+    End18* end18 = &D_80244D74_DFB774[npcID];
+    NpcBlueprint bp;
+    Npc* npc;
+
+    bp.flags = 0x100;
+    bp.initialAnim = end18->initialAnim;
+    bp.onUpdate = NULL;
+    bp.onRender = NULL;
```
Comment (nanaian) [inline src/world/area_end/end_00/DF6A20.c:65]: Can you write this as
```
NpcBlueprint bp = {
    .flags = 0x100,
    .intialAnim = end18->initialAnim,
};
```

Comment (ethteck) [reply src/world/area_end/end_00/DF6A20.c:65]: no, it produces very different cod

**Review by nanaian — APPROVED** [review]

**Pull request merged**